nautobot/utilities/templatetags/helpers.py

```diff
@@ -433,6 +434,24 @@ def settings_or_config(key):
     return get_settings_or_config(key)
 
 
+@library.filter()
+@register.filter()
+def hide_menu(user, admin_only=False):
```

Comment:
Perhaps `should_hide_menu` would be clearer? Or maybe it would be even better to invert the logic of this filter and make it something like `should_show_restricted_ui`?